src/makeInfinityAdder.js

```diff
@@ -3,8 +3,25 @@
 /***
  * @return {function}
  */
+
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  const adder = (value) => {
+    if (value === undefined) {
+      const currentSum = sum;
+
+      sum = 0;
+
+      return currentSum;
+    }
+
+    sum += value;
+
+    return adder;
+  };
+
+  return adder;
```

Comment:
The `makeInfinityAdder` function implementation is correct and adheres to the task description. It correctly accumulates the sum of numbers until it is called without an argument, at which point it returns the accumulated sum and resets it for the next use.